### PR TITLE
DRILL-4354: Remove sessions in anonymous (auth disabled) WebUI access

### PR DESCRIPTION
**Pull request by vkorukanti**

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/DrillRestServer.java [hunk 1]
```diff
@@ -102,4 +112,34 @@ public void dispose(DrillUserPrincipal principal) {
...
+
+    @RequestScoped
+    @Override
+    public DrillUserPrincipal provide() {
+      return new AnonDrillUserPrincipal(workManager.getContext());
+    }
+
+    @Override
+    public void dispose(DrillUserPrincipal principal) {
+      // If this worked it would have been clean to free the resources here, but there are various scenarios
```
Comment (sudheeshkatkam) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/DrillRestServer.java]: Any [specific tickets](https://java.net/jira/browse/JERSEY/)?

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/auth/DrillUserPrincipal.java [hunk 2]
```diff
@@ -60,13 +63,21 @@ public String getName() {
...
+   * @return Return {@link DrillClient} instanced with credentials of this user principal. Returned {@link DrillClient}
+   * must be returned using {@link #recycleDrillClient(DrillClient)} for proper resource cleanup.
    */
-  public DrillClient getDrillClient() {
+  public DrillClient getDrillClient() throws IOException {
     return drillClient;
   }
 
   /**
+   *
```
Comment (sudheeshkatkam) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/auth/DrillUserPrincipal.java]: missing?

**Comment by sudheeshkatkam** [comment]
+1

I have two minor comments. Looks like DRILL-4353 is already committed.

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/DrillRestServer.java (hunk 1 above)
Comment (vkorukanti) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/DrillRestServer.java]: Original issue: https://java.net/jira/browse/JERSEY-2299. It got fixed in 2.7 (we are on 2.8), but I saw few threads that mention it still reproes and not always called when closing the browser (http://jersey.576304.n2.nabble.com/Factory-dispose-is-not-called-td7583326.html). Didn't find any latest JIRAs. When I tried also, it never called after the request is complete. So thought it is not safe to rely on this method to cleanup resources.

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/auth/DrillUserPrincipal.java (hunk 2 above)
Comment (vkorukanti) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/auth/DrillUserPrincipal.java]: added javadoc.

**Pull request closed without merging**